libnavigation-core/src/test/java/com/mapbox/navigation/core/history/MapboxHistoryRecorderTest.kt

```diff
@@ -67,22 +50,15 @@ class MapboxHistoryRecorderTest {
     }
 
     @Test
-    fun `historyRecorder saveHistory logs a warning when it is not initialized`() {
+    fun `historyRecorder stopRecording logs a warning when it is not initialized`() {
         val navigationOptions = navigationOptionsBuilder
-            .historyRecorderOptions(
-                HistoryRecorderOptions.Builder()
-                    .enabled(true)
-                    .build()
-            )
+            .historyRecorderOptions(HistoryRecorderOptions.Builder().build())
             .build()
         val historyRecorder = MapboxHistoryRecorder(navigationOptions, logger)
 
-        val countDownLatch = CountDownLatch(1)
-        historyRecorder.saveHistory { filePath ->
-            countDownLatch.countDown()
-            assertNull(filePath)
```

Comment:
this is needed in order to verify it is called. is it not called anymore?

Reply:
yes, it is not called.
if `historyRecorderHandle` is not initialized, we log it and do nothing. consistent across `startRecording/stopRecording/pushHistory`

Reply:
Ok i think this is fine one way or another and may be a matter of preference. Documentation says it will be called with a `null` path https://github.com/mapbox/mapbox-navigation-android/pull/4587/files#diff-369ecb3399ba6d13c76567951ad1175231d9797b7f95f7c7b6930f0bd9644c49R52

The old way was logging a warning and would always call the callback https://github.com/mapbox/mapbox-navigation-android/pull/4587/files#diff-369ecb3399ba6d13c76567951ad1175231d9797b7f95f7c7b6930f0bd9644c49L72

Personally, my preference is to always call the callback. startRecording and pushHistory do not have callbacks, so there is nothing to be consistent with there. But stopRecording should be consistent with it's own documentation, one way or another

Reply:
> . Documentation says it will be called

yes, but it will be called with `null` only if `startRecording` was not called.
previously `null` was returned when `historyRecorderHandle == null`. It's not true anymore, we just log it